### ISSUE TITLE
emulator with 'nightwatch-android-11' AVD not launching

### ISSUE THREAD

**Issue by richardjimenez52** [nightwatchjs/mobile-helper-tool#8]
I seem to be having issues launching the emulator. I've deleted the emulator folder it downloads and re-downloaded it but still fails to launch. Find screenshot error below:
 nightwatch v2.5.0
nightwatch/mobile-helper 0.1.8
Macbook Apple M1 Pro 

![Screen Shot 2023-01-11 at 4 27 40 PM](https://user-images.githubusercontent.com/102810560/211921603-addebbcb-4cf0-48b7-946c-0f92058670bd.png)

**Comment by garg3133** [comment]
Hi, can you try to launch the emulator manually by running command: `/Users/RichardJi/Library/Android/sdk/emulator/emulator @nightwatch-android-11 -delay-adb`, if that works for you? I'm not sure what could be the reason behind failure here.

**Comment by richardjimenez52** [comment]
Hi @garg3133 , when i run that comment seems I get the following error:

![Screen Shot 2023-01-23 at 12 52 12 PM](https://user-images.githubusercontent.com/102810560/214113048-9724fc2c-a4f8-4747-a66f-57722091c4ce.png)

**Comment by garg3133** [comment]
Thanks! This seems to be a bug. It should have downloaded the `arm64-v8a` Android system image on your system, but instead it downloaded the `x86_64` system image which is meant for the intel laptops.

Can you help me with the value of `process.arch` for your system while running Nodejs? To get this value, run command `node`, and then type `process.arch`.

**Comment by richardjimenez52** [comment]
The process.arch is x64

**Comment by garg3133** [comment]
Got it. It seems you must have changed your arch to work with `x64` since the default arch in Apple M1 is `arm64`.  So, we need to handle this case separately. I'll issue a patch release shortly to handle this case.

**Comment by richardjimenez52** [comment]
@garg3133 I updated to latest version "@nightwatch/mobile-helper": "^0.1.10" and it is still giving me an error and doesn't spawn up.  In the screenshot below I ran the setup command for android and it still failed to spin up the emulator. I also tried to spin it up manually by running `/Users/RichardJi/Library/Android/sdk/emulator/emulator @nightwatch-android-11 -delay-adb` and it failed as well. 

<img width="887" alt="Screen Shot 2023-01-24 at 4 07 45 PM" src="https://user-images.githubusercontent.com/102810560/214415708-ee8620ce-d729-4bca-a648-d2cbe4f72297.png">

**Comment by garg3133** [comment]
That's because you have the system image and corresponding AVD already set up. You need to delete those first. Follow below steps to do so:
```sh
cd /Users/RichardJi/Library/Android/sdk
# delete system image
rm -rf system-images
#delete avd
./cmdline-tools/latest/bin/avdmanager delete avd -n nightwatch-android-11
```

And then re-run the setup. It should setup the AVD correctly this time.

**Comment by richardjimenez52** [comment]
that worked! I was able to spin it up locally and works as expected. But I'm running into one issue, seems the tests are getting skipped after the emulator is up and the file generated in the html reporter is not telling me why either? Not able to run any tests against this android environment. Usually when I seen this happen is because a mismatch of chromedriver version vs google chrome version but this is on an emulator so I'm not sure if it is the same?

this is my configuration:

```
    'android.simulator.chrome': {
      //launch_url: "https://dls-wc-ng-test-app.dls-doc-site.dlas1.ucloud.int/smoke",
      launch_url: "http://localhost:4200/smoke",
      skip_testcases_on_fail: false,
      desiredCapabilities: {
        acceptInsecureCerts: true,
        real_mobile: false,
        avd: 'nightwatch-android-11',
        browserName: 'chrome',
        'goog:chromeOptions': {
          w3c: true,
          args: [
            //'--no-sandbox',
            //'--ignore-certificate-errors',
            //'--allow-insecure-localhost',
            //'--headless'
          ],
          androidPackage: 'com.android.chrome',
          // add the device serial to run tests on, if multiple devices are online
          // Run command: `$ANDROID_HOME/platform-tools/adb devices`
          // androidDeviceSerial: ''
        },
      },

      webdriver: {
        start_process: true,
        server_path: 'chromedriver-mobile/chromedriver',
        cli_args: [
          // --verbose
        ]
      }
    },
```

**Comment by richardjimenez52** [comment]
adding the flag --verbose told me the reason it is not running the tests and its skipping them it is because:
` Error: Unable to create the ChromeDriver process: The specified executable path does not exist: chromedriver-mobile/chromedriver` Do i have to download that binary from somewhere or it comes automatically when running the setup for this npm package?

**Comment by garg3133** [comment]
`chromedriver-mobile/chromedriver` should be created by mobile-helper tool itself. Just run `npx @nightwatch/mobile-helper android --setup` from your project's root dir and choose 'Android Emulator' and 'Google Chrome' for the two questions asked and it should download the required version of chromedriver in the final step.

**Comment by richardjimenez52** [comment]
Seems the reason it was not working by default was because the browser version on the phone was 91 and the chrome version was 109 so I had to automatically download the chrome version 91 and place it inside my root project in the directory chromedriver-mobile/chromedriver. Seems nightwatch-android-11 is running v91 for chrome

**Comment by richardjimenez52** [comment]
@garg3133 is it possible to use another emulator from android studio with a higher chrome version? I created my own pixel 5 on android studio running google api 33 and put its id (name) on my nightwatch config and even downloaded the correct chromedriver version that matches the correct chrome version on pixel 5 but it gave me an error of:

unknown error: Failed to clear data for com.android.chrome on device emulator-5554: cmd: Can't find service: package\n

**Comment by garg3133** [comment]
> Seems nightwatch-android-11 is running v91 for chrome

Ohh, okay! Google must have updated the Chrome version for the Android-11 system image. Would need to update this in this tool as well.

> is it possible to use another emulator from android studio with a higher chrome version?

Yes, that should work as well. I'm not sure why it is giving this error and why is it trying to clear chrome data in the first place. You could look into the chromedriver logs to get more info. If you run your test with `--verbose` flag, you must get a file path like below to the chromedriver logs:
![image](https://user-images.githubusercontent.com/39924567/217756266-f7e7e1bc-4e0c-4e61-a7ea-3a952b80590e.png)

**Comment by garg3133** [comment]
I tried running the test on Android API 33 device myself and I'm able to reproduce this issue.

The initial issue is that it is not waiting for the emulator to boot up completely and starting the test before the boot-up is complete. This seems to be the issue with the system image itself, since we start the device with the `-delay-adb` flag (wait until the boot-up is complete) and it seems that the latest system image is not obeying this flag. You can bypass this issue by starting the emulator yourself and once the boot-up is complete, then run the test.

But even after doing so, I faced another issue in which chromedriver is unable to start the Chrome browser.
![image](https://user-images.githubusercontent.com/39924567/217766304-7a7cc3c0-bd7a-4819-87de-95c435f677bc.png)

And it seems that this issue is also from Google's side: https://github.com/appium/appium/issues/17492

So, I'd recommend to only run your tests up to API level 32 only for now, until these issues are resolved by Google. Or, if you're able to figure out some workarounds, that's also great!

**Comment by richardjimenez52** [comment]
thank you so much for this. I'll be closing it now since all my questions have been resolved.
